tests/mock_crust/merge.rs

```diff
@@ -15,42 +15,75 @@
 // Please review the Licences for the specific language governing permissions and limitations
 // relating to use of the SAFE Network Software.
 
-use rand::distributions::{IndependentSample, Range};
+use rand::Rng;
 use routing::Event;
 use routing::mock_crust::Network;
 use super::{create_connected_nodes_with_cache_until_split, poll_and_resend,
             verify_invariant_for_all_nodes};
 
-#[test]
-fn merge_three_groups_into_one() {
-    // Create a network comprising three groups, one with a prefix `bit_count` of 1 and two with
-    // prefix `bit_count`s of 2.
+// See docs for `create_connected_nodes_with_cache_until_split` for details on `prefix_lengths`.
+fn merge(prefix_lengths: Vec<usize>) {
     let min_group_size = 8;
     let network = Network::new(min_group_size, None);
     let mut rng = network.new_rng();
-    let mut nodes = create_connected_nodes_with_cache_until_split(&network, 3);
+    let mut nodes = create_connected_nodes_with_cache_until_split(&network, prefix_lengths);
     verify_invariant_for_all_nodes(&nodes);
 
-    // Drop nodes from the group with the short prefix until we get a merge event
-    'outer: loop {
-        let range = Range::new(0, nodes.len());
-        let index = range.ind_sample(&mut rng);
-        if unwrap!(nodes[index].routing_table()).our_group_prefix().bit_count() != 1 {
-            continue;
+    // Drop nodes from a group with the shortest prefix until we get a merge event for the empty
+    // Prefix.
+    let mut min_prefix = *unwrap!(nodes[0].routing_table()).our_group_prefix();
+    loop {
+        rng.shuffle(&mut nodes);
+        let mut index = nodes.len();
+        for (i, node) in nodes.iter().enumerate() {
+            let this_prefix = *unwrap!(node.routing_table()).our_group_prefix();
+            if this_prefix.bit_count() < min_prefix.bit_count() {
+                min_prefix = this_prefix;
+                index = i;
+                break;
```

Comment:
So `min_prefix` is not necessarily the shortest prefix now, but just one that is at least shorter than `node[0]`'s?